src/app/DataVersionFilter.h

```diff
@@ -35,6 +36,17 @@ struct DataVersionFilter
         return (mEndpointId != kInvalidEndpointId) && (mClusterId != kInvalidClusterId) && (mDataVersion.HasValue());
     }
 
+    bool operator==(const DataVersionFilter & aOther) const
+    {
+        return mEndpointId == aOther.mEndpointId && mClusterId == aOther.mClusterId &&
+            mDataVersion.Value() == aOther.mDataVersion.Value();
```

Comment:
Is there a reason this is not just comparing `mDataVersion == aOther.mDataVersion`?

Reply:
good catch, updated, thanks